stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/model/response/SocketErrorResponse.kt

```diff
@@ -34,6 +34,14 @@ internal data class SocketErrorResponse(
         val StatusCode: Int = -1,
         val duration: String = "",
         val exception_fields: Map<String, String> = mapOf(),
-        val more_info: String = ""
-    )
+        val more_info: String = "",
+        val details: List<ErrorDetail>
+    ) {
+
+        @JsonClass(generateAdapter = true)
+        data class ErrorDetail(
```

Comment:
Instead of having this as a nested class and using the moshi adapter dependency and generating more stuff which can be overhead, can we just pull this out as a separate class and object and then parse it as an array or list by default, do we need adapters for that?